### PR TITLE
37974 - add key components to the modules introduction

### PR DESCRIPTION
**Pull request by theo-ritense**

### CODE REVIEW

File: introduction/modules/modules.md [hunk 1]
```diff
@@ -15,41 +18,51 @@ Connectors are Valtimo configurable components that can be used to connect to ex
...
-Contract contains certain interfaces and events that Valtimo uses in several other modules. Apart from this, contract
-sanitizes endpoint output when exceptions occur. A liquibase runner is available that allows other modules to define
-liquibase configuration that is run on startup. Finally, Contract includes several helper classes, like the
-`QueryDialectHelper` bean that can be used in criteria queries to add JSON column clauses to the query.
+Contract contains certain interfaces and events that Valtimo uses in several other modules. Consider it the 'glue'
+between a module and Core.
+
+Apart from this, contract sanitizes endpoint output when exceptions occur. A liquibase runner is available that allows
+other modules to define liquibase configuration that is run on startup. Finally, Contract includes several helper
+classes, like the `QueryDialectHelper` bean that can be used in criteria queries to add JSON column clauses to the
```
Comment (ivo-ritense) [inline introduction/modules/modules.md:30]:
```suggestion
that can be used in criteria queries to add JSON column clauses to the
query.
```

that can be used in criteria queries to call DBMS specific functions in a generic way (e.g. finding JSON data).

File: introduction/modules/modules.md [hunk 2]
```diff
@@ -15,41 +18,51 @@ Connectors are Valtimo configurable components that can be used to connect to ex
...
+It also includes choice fields. A generic way to create lists of values, that can be used by the application. Some
+security features used to control access to any Valtimo endpoints are also part of this module.
 
 ### Documents
 
-A document is the starting point of the case and the main data storage for that case. One or more processes can be
-linked to a case, which allows for the document to be created when the case is started. The content of the document is 
-defined in a document definition, in JSON schema. Users interact with the document via user tasks; document data can be 
-loaded into task forms, and user input can be stored in the document after completing a task.
+A document is the main data storage entity that can contain data for one or more processes. It can be linked to a case,
```
Comment (ivo-ritense) [inline introduction/modules/modules.md:43]: This is not completely correct. technically a document is the same as a case right now so it cannot be 'linked' to a case. 

```suggestion
A document is the main data storage entity that contains data of a case. Processes can act upon and change the data that is contained in the document. 
```

File: introduction/modules/modules.md [hunk 3]
```diff
@@ -15,41 +18,51 @@ Connectors are Valtimo configurable components that can be used to connect to ex
...
 ### Core
 
-Core contains and builds upon the Camunda process engine that enables Valtimo to run processes. Several endpoints and
-functions are available that provide task data and process data. It also includes choice fields. A generic way to create
-lists of values, that can be used by the application. Some security features used to control access to any Valtimo
-endpoints are also part of this module.
+Core is the main module of Valtimo. It contains and builds upon the Camunda process engine that enables Valtimo to run
+processes. Several endpoints and functions are available that provide task data and process data.
+
+It also includes choice fields. A generic way to create lists of values, that can be used by the application. Some
```
Comment (marijnritense) [inline introduction/modules/modules.md:38]: Maybe provide an example as to what choice fields can be used for?

File: introduction/modules/modules.md [hunk 4]
```diff
@@ -74,102 +87,112 @@ to be submitted to the submission endpoint in this module.
...
+Process document facilitates the link between [documents](#documents) and BPMN processes. In order to use a process
+with a document, the process needs to be linked to a document. Process document also provides access of document data
 from inside a process.
 
 ### Resource
 
-Resource is a module that describes how resources can be uploaded and downloaded. This implemented by other modules,
-e.g. [OpenZaak resource](#openzaak-resource).
+Resource is a module that describes how files can be uploaded and downloaded. This implemented by other modules,
+such as the [Temporary resource storage](#temporary-resource-storage) and [OpenZaak resource](#openzaak-resource).
```
Comment (marijnritense) [inline introduction/modules/modules.md:151]: I don't think the t in temporary needs to be capitalized.

File: introduction/modules/modules.md (hunk 3 above)
Comment (theo-ritense) [reply introduction/modules/modules.md]: Not sure if choice fields are so important that we need to mention it at all actually. However, it was already there and I applied your suggestion.

File: introduction/modules/modules.md [hunk 5]
```diff
@@ -74,102 +88,112 @@ to be submitted to the submission endpoint in this module.
...
 another module.
 
 ### Mandrill
 
-Mailchimp is a SaaS solution for sending emails. Mailchimp is the name of the 
-[transactional mail addon](https://mailchimp.com/features/transactional-email/) that Mailchimp provides. The Valtimo
-Mandrill module provides a mail implementation that uses Mandrill to send emails.
+Mailchimp is a SaaS solution for sending emails. Mandrill is the name of
+the [transactional mail addon](https://mailchimp.com/features/transactional-email/) that Mailchimp provides. The Valtimo
+Mandrill module provides a mail implementation of [Mail](#mail) that Valtimo uses to send emails..
```
[the remark this answers is not in the document]
Comment (theo-ritense) [reply introduction/modules/modules.md:122]: good catch.

Comment (theo-ritense) [inline introduction/modules/modules.md:122]:
```suggestion
Mandrill module provides a mail implementation of [Mail](#mail) that Valtimo uses to send emails.
```

**Review by ivo-ritense — APPROVED** [review]

**Pull request merged**